examples/query/react/basic/src/test/test-utils.tsx

```diff
@@ -0,0 +1,45 @@
+import { render as rtlRender } from '@testing-library/react'
+import type { RenderOptions } from '@testing-library/react'
+import React, { PropsWithChildren } from 'react'
+import { Provider } from 'react-redux'
+import { setupStore } from '../store'
+import type { AppStore, RootState } from '../store'
+import type { PreloadedState } from '@reduxjs/toolkit'
+
+// This type interface extends the default options for render from RTL, as well
+// as allows the user to specify other things such as initialState, store. For
+// future dependencies, such as wanting to test with react-router, you can extend
+// this interface to accept a path and route and use those in a <MemoryRouter />
+interface IWithStoreOptions extends Omit<RenderOptions, 'queries'> {
+  initialState?: PreloadedState<RootState>
+  store?: AppStore
+}
+
+function render(
```

Comment:
@dggsax @markerikson @phryneas @msutkowski 

I'd like feedback from as many people as possible about these particular points:

1. I'd personally prefer to explicitly *not* call this `render`, and make it something like `renderWithProviders` instead

2. I'd also personally prefer to not re-export everything from `RTL` in this file - I don't think saving a line when importing in this case is worth the obfuscation about the actual source of your imports. At a glance in the testing file, you won't know which ones actually have customized behaviour or not when they're all imported from a custom `test-utils` file, and particularly when the ones that *are* customized have hi-jacked the original names.

FWIW for point 2, I've also seen cases before with jetbrains having problems retaining the type information when re-exported with that pattern

Reply:
Agree on both.
Especially since `hooks-testing-library` has similarly named exports with slightly On 2: differing functionality. So having the user import stuff from a source they don't know 100% might be counterproductive.

Reply:
So I just pushed a commit to rename it to `renderWithProviders()` as suggested.

As for part two for re-exporting from a library, then in that case I should probably go ahead and remove the line `export * from 'msw'` line from `./src/test/server/index.ts` to be consistent.